helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java

```diff
@@ -145,6 +148,35 @@ public void testPostRequestFormat() throws IOException {
     Assert.assertEquals(json.get("data").asText(), "{}");
   }
 
+  @Test
+  public void testGetInstanceStoppableCheckSmallTimeout() throws Exception {
+    // Reset the INSTANCE in CustomRestClientFactory to create new one with new TimeOut
+    Field instance = CustomRestClientFactory.class.getDeclaredField("INSTANCE");
+    instance.setAccessible(true);
+    instance.set(null, null);
+
+    // Set 1 ms to cause timeout for http requests
+    System.setProperty(RestSystemPropertyKeys.HTTP_TIMEOUT_MS, "1");
+    // a popular echo server that echos all the inputs
+    // TODO: add a mock rest server
+    final String echoServer = "http://httpbin.org/post";
+    CustomRestClient _customRestClient = CustomRestClientFactory.get();
+    boolean timeoutExceptionHappened = false;
+    try {
+      _customRestClient.getInstanceStoppableCheck(echoServer, Collections.emptyMap());
+    } catch (ConnectTimeoutException e) {
+      // Since the timeout is so small, we are expecting to get ConnectTimeoutException
```

Comment:
I am not sure if I am convinced by this. What if it is fast enough to complete the request in 1 ms? It may not happen, but theoretically it could. This is relying on possibility of timeout..

Reply:
Changed the test completely. It is using mock now.